### PR TITLE
Added Azure.AppConfig.GeoReplication

### LINKED ISSUE #1688: Use geo-replication for App Configuration stores
# Rule request

## Suggested rule change

Enable replication for configuration stores to ensure that stores are **resilient** to region outages.

A replica is created by the `Microsoft.AppConfiguration/configurationStores/replicas` sub-resource.

This is a preview feature.

Currently this feature is not available on the free tier, so the rule should ignore free tier stores.

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.AppConfiguration/configurationStores**

## Additional context

[Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.appconfiguration/configurationstores/replicas)
[Geo replication](https://learn.microsoft.com/azure/azure-app-configuration/concept-geo-replication)

Related rules include:

- https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ACR.GeoReplica/

### PR DESCRIPTION
**Pull request by bengeset96**
## PR Summary

Fixes #1688

Added Azure.AppConfig.GeoReplication rule. 

This rule is aimed at app configuration stores, ensuring that they have one or more replicas providing better resiliency to region outages.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by bengeset96** [comment]
@BernieWhite 

Should be ready for review.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 Great work. Just a few minor tweaks to make this even better. If you have any questions please let me know.

File: src/PSRule.Rules.Azure/rules/Azure.AppConfig.Rule.ps1
Comment (bengeset96) [reply src/PSRule.Rules.Azure/rules/Azure.AppConfig.Rule.ps1]: I agree. This was not optimal from my side and I will adjust accordingly.

**Comment by bengeset96** [comment]
@BernieWhite 

I have refactored the rule now. Do you mind to take a look on it?

`2022_09` baseline included, please inform if should be adjusted to `2022_12`.

File: src/PSRule.Rules.Azure/rules/Azure.AppConfig.Rule.ps1
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.AppConfig.Rule.ps1:36]:
```suggestion
    $Assert.Greater($replicas, '.', 0).Reason($LocalizedData.ReplicaInSecondaryNotFound).PathPrefix('resources')
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 One minor tweak. Otherwise all good. We're fine to go with `2022_09` for this rule. I'll merge lock in the baseline after we merge this rule.

**Comment by bengeset96** [comment]
Done :)

**Review by BernieWhite — APPROVED** [review]
Thanks @bengeset96 great work. All good to merge.

**Pull request merged**
